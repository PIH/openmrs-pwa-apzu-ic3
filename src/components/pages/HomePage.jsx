import React from 'react';
import { Button, ButtonGroup, Glyphicon } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import RequiredScreenings from '../../screening/RequiredScreenings';

const HomePage = () => (
  <div className="App">

    <b>Home Page</b>
    <br />
    <br />
    <div>
      <Link to="/searchPatient">
        <ButtonGroup>
          <Button bsSize="large" >
            <Glyphicon glyph="check" /> Search Patient
          </Button>
        </ButtonGroup>
      </Link>
      <Link to="/screening/bloodPressureQueue">
        <ButtonGroup>
          <Button bsSize="large" >
            <Glyphicon glyph="check" /> Blood Pressure Screening
          </Button>
        </ButtonGroup>
      </Link>
      <Link to="/screening/nutritionQueue">
        <ButtonGroup>
          <Button bsSize="large" >
            <Glyphicon glyph="check" /> Nutrition Screening
          </Button>
        </ButtonGroup>
      </Link>
      <RequiredScreenings />
    </div>

  </div>
);

export default HomePage;
