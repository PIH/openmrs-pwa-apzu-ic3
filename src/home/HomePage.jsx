import React from 'react';
import { connect } from "react-redux";
import { LOCATION_TYPES } from '../constants';
import patientActions from '../patient/patientActions';
import homeImage from "../assets/images/Malawi_0216_HMHC-6weekANC_JDrake_033-webbig.jpg";
import '../assets/css/background.css';

class HomePage extends React.Component {

  componentDidMount() {
    this.props.dispatch(patientActions.clearPatientSelected());
  }

  render() {
    return (
      <div
        className="background"
        style={{backgroundImage: 'url(' + homeImage + ')',}}
      />

    )
  }
}

const mapStateToProps = (state) => {
  return {
    location: state.openmrs.session.sessionLocation ? state.openmrs.session.sessionLocation.uuid : LOCATION_TYPES.UnknownLocation
  };
};


export default connect(mapStateToProps)(HomePage);
