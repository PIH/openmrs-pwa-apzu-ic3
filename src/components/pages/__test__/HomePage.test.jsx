import React from 'react'
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import HomePage from '../HomePage'
import configureMockStore from 'redux-mock-store'
import { BrowserRouter, Route } from 'react-router-dom';

let props, mountedComponent, store;

const mockStore = configureMockStore();
const homePage = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(
      <BrowserRouter><Route><HomePage {...props} /></Route></BrowserRouter>)
  }
  return mountedComponent;
};

describe('Component: HomePage', () => {
  beforeEach(() => {
    props = {};
    mountedComponent = undefined;
  });

  it('renders properly', () => {
    expect(toJson(homePage())).toMatchSnapshot();
    expect(homePage().find(HomePage).length).toBe(1);
    // expect(homePage().props('bsSize').length).toBe(1);
  });

});

