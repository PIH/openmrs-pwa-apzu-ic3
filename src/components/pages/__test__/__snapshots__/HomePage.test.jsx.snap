// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: HomePage renders properly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Route>
    <HomePage />
  </Route>
</Router>
`;
